beets/autotag/mb.py

```diff
@@ -429,6 +429,28 @@ def match_album(artist, album, tracks=None):
     if tracks is not None:
         criteria['tracks'] = six.text_type(tracks)
 
+    # Additional search cues from existing metadata.
+    if extra_tags is not None:
+        catalognum = extra_tags.get('catalognum', '').lower().strip()
+        if re.search(u'^[0-9a-z -]+$', catalognum):
```

Comment:
This regex filter seems like a reasonably good idea, but just checking: is this based on a guess about what most catalog numbers look like, or is this based on some manner of MusicBrainz guidance?

Reply:
Honestly, I don't remember why I added that regex check. I don't think there's any reason to safeguard against bad catalog numbers in particular when other fields aren't validated, so I just removed this.

Note that we do still need the special handling of `catno.replace(u' ', '')` since MediaBrainz appears to be unable to match catalog numbers containing spaces.